Origin: EmberBSD (AI-assisted). Build these consumers with the one Ruby, Go,
Lua, MySQL client, FFmpeg, PCRE and Qt version in the export instead of an old
version.
Umbrello 26.08 builds with Qt 6 and KDE Frameworks 6 through BUILD_WITH_QT6, and
then suffixes its programs and data directory with 6 instead of 5.
ZoneMinder 1.38 already looks for PCRE2, so with the old PCRE it lost regex
support silently.
VLC's Lua scripts used math.pow and assigned to a for-loop variable; Lua 5.5
allows neither, so the scripts compute the power with ^ and use new locals.
OpenSceneGraph's own Lua finders stop at Lua 5.2, so it gets the library and
headers directly; its Lua plugin needs no source change.

--- a/devel/Guitar/Makefile
+++ b/devel/Guitar/Makefile
@@ -11,7 +11,7 @@ HOMEPAGE=	https://github.com/soramimi/Guitar/
 COMMENT=	Git GUI Client
 LICENSE=	gnu-gpl-v2
 
-TOOL_DEPENDS+=	ruby34-[0-9]*:../../lang/ruby34
+TOOL_DEPENDS+=	${RUBY_BASE}>=${RUBY_VERSION}:${RUBY_SRCDIR}
 
 USE_TOOLS+=	pkg-config
 USE_LANGUAGES=	c c++
@@ -32,11 +32,12 @@ do-configure:
 	cd ${WRKSRC}/filetype && ${QTDIR}/bin/qmake liboniguruma.pro
 	cd ${WRKSRC}/filetype && ${SETENV} ${MAKE_ENV} ${MAKE}
 	cd ${WRKSRC} && ${QTDIR}/bin/qmake Guitar.pro
-	${LN} -s ${PREFIX}/bin/ruby34 ${BUILDLINK_DIR}/bin/ruby
+	${LN} -s ${TOOLBASE}/bin/${RUBY_NAME} ${BUILDLINK_DIR}/bin/ruby
 
 post-install:
 	${INSTALL_PROGRAM} ${WRKSRC}/_bin/Guitar ${DESTDIR}${PREFIX}/bin
 
+.include "../../lang/ruby/rubyversion.mk"
 .include "../../graphics/qt6-qtsvg/buildlink3.mk"
 .include "../../x11/qt6-qtbase/buildlink3.mk"
 .include "../../mk/bsd.pkg.mk"
--- a/devel/umbrello/Makefile
+++ b/devel/umbrello/Makefile
@@ -1,25 +1,31 @@
 # $NetBSD: Makefile,v 1.107 2026/09/06 11:02:45 markd Exp $
 
 DISTNAME=	umbrello-${KAPPSVER}
+PKGREVISION=	1
 CATEGORIES=	devel
 
 HOMEPAGE=	https://kde.org/applications/development/umbrello/
 COMMENT=	UML Modeller
 LICENSE=	gnu-lgpl-v2 AND gnu-gpl-v2 AND gnu-fdl-v1.2
 
-STILL_KF5=	yes
 .include "../../meta-pkgs/kde/applications.mk"
 
-TOOL_DEPENDS+=	kdoctools>=5.19.0:../../devel/kdoctools
+TOOL_DEPENDS+=	kf6-kdoctools>=6.18.0:../../devel/kf6-kdoctools
 
 CMAKE_CONFIGURE_ARGS+=	-DBUILD_APIDOC=no
+CMAKE_CONFIGURE_ARGS+=	-DBUILD_WITH_QT6=ON
 
 USE_TOOLS+=	msgmerge
 
 .include "../../devel/gettext-lib/buildlink3.mk"
-.include "../../devel/ktexteditor/buildlink3.mk"
+.include "../../devel/kf6-kcrash/buildlink3.mk"
+.include "../../devel/kf6-ktexteditor/buildlink3.mk"
 .include "../../graphics/hicolor-icon-theme/buildlink3.mk"
+.include "../../graphics/kf6-kiconthemes/buildlink3.mk"
+.include "../../graphics/qt6-qtsvg/buildlink3.mk"
 .include "../../sysutils/desktop-file-utils/desktopdb.mk"
 .include "../../textproc/libxslt/buildlink3.mk"
-.include "../../x11/qt5-qtbase/buildlink3.mk"
+.include "../../x11/kf6-kwindowsystem/buildlink3.mk"
+.include "../../x11/kf6-kxmlgui/buildlink3.mk"
+.include "../../x11/qt6-qtbase/buildlink3.mk"
 .include "../../mk/bsd.pkg.mk"
--- a/devel/umbrello/PLIST
+++ b/devel/umbrello/PLIST
@@ -1,7 +1,7 @@
 @comment $NetBSD: PLIST,v 1.12 2026/09/06 11:02:45 markd Exp $
-bin/po2xmi5
-bin/umbrello5
-bin/xmi2pot5
+bin/po2xmi6
+bin/umbrello6
+bin/xmi2pot6
 share/applications/org.kde.umbrello.desktop
 share/doc/HTML/ca/umbrello/apphelp/index.cache.bz2
 share/doc/HTML/ca/umbrello/apphelp/index.docbook
@@ -335,50 +335,50 @@
 share/locale/zh_TW/LC_MESSAGES/umbrello.mo
 share/locale/zh_TW/LC_MESSAGES/umbrello_kdevphp5.mo
 share/metainfo/org.kde.umbrello.appdata.xml
-share/umbrello5/common.ent
-share/umbrello5/docbook2xhtml.xsl
-share/umbrello5/headings/heading.adb
-share/umbrello5/headings/heading.ads
-share/umbrello5/headings/heading.as
-share/umbrello5/headings/heading.cpp
-share/umbrello5/headings/heading.cs
-share/umbrello5/headings/heading.d
-share/umbrello5/headings/heading.h
-share/umbrello5/headings/heading.idl
-share/umbrello5/headings/heading.java
-share/umbrello5/headings/heading.js
-share/umbrello5/headings/heading.php
-share/umbrello5/headings/heading.pm
-share/umbrello5/headings/heading.py
-share/umbrello5/headings/heading.rb
-share/umbrello5/headings/heading.sql
-share/umbrello5/headings/heading.xsd
-share/umbrello5/layouts/activity-export.desktop
-share/umbrello5/layouts/activity-vertical.desktop
-share/umbrello5/layouts/class-export.desktop
-share/umbrello5/layouts/class-horizontal.desktop
-share/umbrello5/layouts/class-vertical.desktop
-share/umbrello5/layouts/component-horizontal.desktop
-share/umbrello5/layouts/component-vertical.desktop
-share/umbrello5/layouts/entityrelationship-export.desktop
-share/umbrello5/layouts/entityrelationship-horizontal.desktop
-share/umbrello5/layouts/entityrelationship-vertical.desktop
-share/umbrello5/layouts/object-export.desktop
-share/umbrello5/layouts/object-horizontal.desktop
-share/umbrello5/layouts/object-vertical.desktop
-share/umbrello5/layouts/state-export.desktop
-share/umbrello5/layouts/state-horizontal.desktop
-share/umbrello5/layouts/state-vertical.desktop
-share/umbrello5/layouts/usecase-default.desktop
-share/umbrello5/simple4125/sdbcent.mod
-share/umbrello5/simple4125/sdbhier.mod
-share/umbrello5/simple4125/sdbpool.mod
-share/umbrello5/simple4125/sdocbook-custom.dtd
-share/umbrello5/simple4125/sdocbook.css
-share/umbrello5/simple4125/sdocbook.dtd
-share/umbrello5/simple4125/sdocbookref-custom.dtd
-share/umbrello5/simple4125/sdocbookref.dtd
-share/umbrello5/simple4125/sinclist.mod
-share/umbrello5/xmi.css
-share/umbrello5/xmi1docbook.xsl
-share/umbrello5/xmi2docbook.xsl
+share/umbrello6/common.ent
+share/umbrello6/docbook2xhtml.xsl
+share/umbrello6/headings/heading.adb
+share/umbrello6/headings/heading.ads
+share/umbrello6/headings/heading.as
+share/umbrello6/headings/heading.cpp
+share/umbrello6/headings/heading.cs
+share/umbrello6/headings/heading.d
+share/umbrello6/headings/heading.h
+share/umbrello6/headings/heading.idl
+share/umbrello6/headings/heading.java
+share/umbrello6/headings/heading.js
+share/umbrello6/headings/heading.php
+share/umbrello6/headings/heading.pm
+share/umbrello6/headings/heading.py
+share/umbrello6/headings/heading.rb
+share/umbrello6/headings/heading.sql
+share/umbrello6/headings/heading.xsd
+share/umbrello6/layouts/activity-export.desktop
+share/umbrello6/layouts/activity-vertical.desktop
+share/umbrello6/layouts/class-export.desktop
+share/umbrello6/layouts/class-horizontal.desktop
+share/umbrello6/layouts/class-vertical.desktop
+share/umbrello6/layouts/component-horizontal.desktop
+share/umbrello6/layouts/component-vertical.desktop
+share/umbrello6/layouts/entityrelationship-export.desktop
+share/umbrello6/layouts/entityrelationship-horizontal.desktop
+share/umbrello6/layouts/entityrelationship-vertical.desktop
+share/umbrello6/layouts/object-export.desktop
+share/umbrello6/layouts/object-horizontal.desktop
+share/umbrello6/layouts/object-vertical.desktop
+share/umbrello6/layouts/state-export.desktop
+share/umbrello6/layouts/state-horizontal.desktop
+share/umbrello6/layouts/state-vertical.desktop
+share/umbrello6/layouts/usecase-default.desktop
+share/umbrello6/simple4125/sdbcent.mod
+share/umbrello6/simple4125/sdbhier.mod
+share/umbrello6/simple4125/sdbpool.mod
+share/umbrello6/simple4125/sdocbook-custom.dtd
+share/umbrello6/simple4125/sdocbook.css
+share/umbrello6/simple4125/sdocbook.dtd
+share/umbrello6/simple4125/sdocbookref-custom.dtd
+share/umbrello6/simple4125/sdocbookref.dtd
+share/umbrello6/simple4125/sinclist.mod
+share/umbrello6/xmi.css
+share/umbrello6/xmi1docbook.xsl
+share/umbrello6/xmi2docbook.xsl
--- a/editors/mle/Makefile
+++ b/editors/mle/Makefile
@@ -1,7 +1,7 @@
 # $NetBSD: Makefile,v 1.7 2026/09/02 19:01:50 wiz Exp $
 
 DISTNAME=	mle-1.8.1
-PKGREVISION=	1
+PKGREVISION=	2
 CATEGORIES=	editors
 MASTER_SITES=	${MASTER_SITE_GITHUB:=adsr/}
 GITHUB_TAG=	v${PKGVERSION_NOREV}
@@ -23,9 +23,9 @@
 INSTALLATION_DIRS+=	bin ${PKGMANDIR}/man1
 
 SUBST_CLASSES+=		lua
-SUBST_MESSAGE.lua=	rewrite lua directory directives
-SUBST_FILES.lua=	mle.h
-SUBST_SED.lua=		-e 's,lua5.4/,lua-5.4/,g'
+SUBST_MESSAGE.lua=	use the default Lua headers and library
+SUBST_FILES.lua=	Makefile mle.h
+SUBST_SED.lua=		-e 's,lua5\.4/,,g' -e 's,-llua5\.4,-llua,g'
 SUBST_STAGE.lua=	pre-build
 
 SUBST_CLASSES+=		makefile
@@ -38,6 +38,6 @@
 	${INSTALL_MAN} ${WRKSRC}/mle.1 ${DESTDIR}${PREFIX}/${PKGMANDIR}/man1
 
 .include "../../devel/pcre2/buildlink3.mk"
-.include "../../lang/lua54/buildlink3.mk"
+.include "../../lang/lua/buildlink3.mk"
 .include "../../devel/uthash/buildlink3.mk"
 .include "../../mk/bsd.pkg.mk"
--- a/graphics/osg/Makefile
+++ b/graphics/osg/Makefile
@@ -1,7 +1,7 @@
 # $NetBSD: Makefile,v 1.213 2026/09/02 19:02:49 wiz Exp $
 
 DISTNAME=	osg-3.6.5
-PKGREVISION=	86
+PKGREVISION=	87
 CATEGORIES=	graphics devel
 MASTER_SITES=	${MASTER_SITE_GITHUB:=openscenegraph/}
 GITHUB_PROJECT=	OpenSceneGraph
@@ -40,6 +40,9 @@
 CMAKE_CONFIGURE_ARGS+=		-DCMAKE_DISABLE_FIND_PACKAGE_COLLADA=TRUE
 CMAKE_CONFIGURE_ARGS+=		-DCMAKE_DISABLE_FIND_PACKAGE_FBX=TRUE
 CMAKE_CONFIGURE_ARGS+=		-DOSG_USE_LOCAL_LUA_SOURCE=OFF
+# The bundled finders know Lua 5.1 and 5.2 only; the Lua plugin builds with 5.5.
+CMAKE_CONFIGURE_ARGS+=		-DLUA_INCLUDE_DIR=${BUILDLINK_PREFIX.lua}/${LUA_INCDIR}
+CMAKE_CONFIGURE_ARGS+=		-DLUA_LIBRARY:STRING=-llua
 CMAKE_CONFIGURE_ARGS+= 	-DCMAKE_POLICY_VERSION_MINIMUM=3.5
 
 PLIST_VARS+=	quartz notdarwin
@@ -58,7 +61,7 @@
 .include "../../graphics/jasper/buildlink3.mk"
 .include "../../graphics/openexr/buildlink3.mk"
 .include "../../graphics/tiff/buildlink3.mk"
-.include "../../lang/lua52/buildlink3.mk"
+.include "../../lang/lua/buildlink3.mk"
 .include "../../multimedia/ffmpeg4/buildlink3.mk"
 .include "../../www/curl/buildlink3.mk"
 .if ${OPSYS} != "Darwin"
--- a/multimedia/vlc/Makefile
+++ b/multimedia/vlc/Makefile
@@ -1,7 +1,7 @@
 # $NetBSD: Makefile,v 1.325 2026/09/02 19:03:50 wiz Exp $
 
 DISTNAME=		vlc-3.0.23
-PKGREVISION=		8
+PKGREVISION=		9
 CATEGORIES=		multimedia
 MASTER_SITES=		https://download.videolan.org/pub/videolan/vlc/${PKGVERSION_NOREV}/
 EXTRACT_SUFX=		.tar.xz
@@ -35,7 +35,7 @@
 .include "options.mk"
 
 CONFIGURE_ENV+=		BUILDCC=${CC:Q}
-CONFIGURE_ENV+=		LUAC=${PREFIX}/bin/luac5.2
+CONFIGURE_ENV+=		LUAC=${LUA_COMPILER}
 
 # Avoid codec duplication with ffmpeg, etc
 
@@ -215,7 +215,7 @@
 .include "../../graphics/freetype2/buildlink3.mk"
 .include "../../graphics/hicolor-icon-theme/buildlink3.mk"
 .include "../../graphics/png/buildlink3.mk"
-.include "../../lang/lua52/buildlink3.mk"
+.include "../../lang/lua/buildlink3.mk"
 .include "../../misc/libcdio/buildlink3.mk"
 .include "../../mk/jpeg.buildlink3.mk"
 .include "../../multimedia/libass/buildlink3.mk"
--- a/multimedia/vlc/distinfo
+++ b/multimedia/vlc/distinfo
@@ -11,3 +11,5 @@
 SHA1 (patch-modules_access_http_Makefile.am) = d91904c6cba1a75a6872d042e4fe87ed9ce65a22
 SHA1 (patch-modules_services__discovery_mtp.c) = 5b5b9b341c71a3273178493ae55fccb583f5bd59
 SHA1 (patch-share_Makefile.am) = 4f126948c73159d06aef323b47a702442b5981e8
+SHA1 (patch-share_lua_intf_modules_httprequests.lua) = a9f4505d17818e161ee71df9ff68aad1a8687ed8
+SHA1 (patch-share_lua_sd_fmc.lua) = 88d9fb65c002c45034d799fb8ca4696f04f0791a
--- /dev/null
+++ b/multimedia/vlc/patches/patch-share_lua_intf_modules_httprequests.lua
@@ -0,0 +1,15 @@
+$NetBSD$
+
+Lua 5.3 and later have no math.pow without compatibility options.
+
+--- share/lua/intf/modules/httprequests.lua.orig
++++ share/lua/intf/modules/httprequests.lua
+@@ -34,7 +34,7 @@ function round(what, precision)
+         what = common.us_tonumber(what)
+     end
+     if type(what) == "number" then
+-        return math.floor(what*math.pow(10,precision)+0.5) / math.pow(10,precision)
++        return math.floor(what*10^precision+0.5) / 10^precision
+     end
+     return nil
+ end
--- /dev/null
+++ b/multimedia/vlc/patches/patch-share_lua_sd_fmc.lua
@@ -0,0 +1,15 @@
+$NetBSD$
+
+Lua 5.5 for-loop variables are read-only; declare the match results local.
+
+--- share/lua/sd/fmc.lua.orig
++++ share/lua/sd/fmc.lua
+@@ -32,7 +32,7 @@ function main()
+         if tonumber( show_node.children_map["songcount"][1].children[1] ) > 0 then
+             local songs_node = node:add_node( {title="Songs"} )
+             for _, song_node in ipairs( show_node.children_map["songs"][1].children ) do
+-                _, _, artist, title = string.find( song_node.children_map["name"][1].children[1], "(.+)%s*-%s*(.+)" )
++                local _, _, artist, title = string.find( song_node.children_map["name"][1].children[1], "(.+)%s*-%s*(.+)" )
+                 local rank = song_node.children_map["rank"][1].children[1]
+                 if rank ~= nil then
+                     rank = "Rank: " .. rank
--- a/net/gnunet/buildlink3.mk
+++ b/net/gnunet/buildlink3.mk
@@ -27,7 +27,7 @@ pkgbase:= gnunet
 .endif
 
 .if ${PKG_BUILD_OPTIONS.gnunet:Mmysql}
-.include "../../databases/mysql80-client/buildlink3.mk"
+.include "../../mk/mysql.buildlink3.mk"
 .endif
 
 .if ${PKG_BUILD_OPTIONS.gnunet:Mpgsql}
--- a/net/grilo-plugins/Makefile
+++ b/net/grilo-plugins/Makefile
@@ -1,7 +1,7 @@
 # $NetBSD: Makefile,v 1.67 2026/09/02 19:03:54 wiz Exp $
 
 DISTNAME=	grilo-plugins-0.3.16
-PKGREVISION=	9
+PKGREVISION=	10
 CATEGORIES=	net gnome
 MASTER_SITES=	${MASTER_SITE_GNOME:=sources/${PKGBASE}/${PKGVERSION_NOREV:R}/}
 EXTRACT_SUFX=	.tar.xz
@@ -22,7 +22,8 @@
 .include "../../devel/gperf/buildlink3.mk"
 .include "../../devel/meson/build.mk"
 .include "../../databases/sqlite3/buildlink3.mk"
-.include "../../lang/lua53/buildlink3.mk"
+LUA_VERSIONS_ACCEPTED=	55 54 53
+.include "../../lang/lua/buildlink3.mk"
 .include "../../multimedia/gstreamer1/buildlink3.mk"
 .include "../../net/grilo/buildlink3.mk"
 .include "../../net/libgdata/buildlink3.mk"
--- a/net/qrcp/Makefile
+++ b/net/qrcp/Makefile
@@ -16,8 +16,6 @@ LICENSE=	mit
 WRKSRC=		${WRKDIR}/${PKGNAME_NOREV}
 USE_LANGUAGES=	c
 
-TOOL_DEPENDS+=	go118>=1.18.0:../../lang/go118
-
 GO_BUILD_PATTERN=	.
 
 .include "./go-modules.mk"
--- a/security/zoneminder/Makefile
+++ b/security/zoneminder/Makefile
@@ -8,7 +8,7 @@
 NAME_DIST=	ZoneMinder
 NAME_LOWER=	zoneminder
 PKGNAME=	${DISTNAME:S/${NAME_DIST}-/${NAME_LOWER}-/}
-PKGREVISION=	3
+PKGREVISION=	4
 DISTNAME=	${NAME_DIST}-${VERSION}
 WRKSRC=		${WRKDIR}/${NAME_LOWER}-${GITHUB_TAG}
 CATEGORIES=	security
@@ -149,11 +149,10 @@
 PTHREAD_AUTO_VARS=	yes
 .include "../../mk/pthread.buildlink3.mk"
 
-# \todo Move to higher ffmpeg.
-.include "../../multimedia/ffmpeg8/buildlink3.mk"
+.include "../../multimedia/ffmpeg9/buildlink3.mk"
 .for lib in AVFORMAT AVCODEC AVDEVICE AVUTIL SWSCALE SWRESAMPLE
-CMAKE_CONFIGURE_ARGS+=	-D${lib}_INCLUDE_DIR=${BUILDLINK_PREFIX.ffmpeg8}/include/ffmpeg8
-CMAKE_CONFIGURE_ARGS+=	-D${lib}_LIBRARIES=${BUILDLINK_PREFIX.ffmpeg8}/lib/ffmpeg8/lib${lib:tl}.${SHLIB_EXT}
+CMAKE_CONFIGURE_ARGS+=	-D${lib}_INCLUDE_DIR=${BUILDLINK_PREFIX.ffmpeg9}/include/ffmpeg9
+CMAKE_CONFIGURE_ARGS+=	-D${lib}_LIBRARIES=${BUILDLINK_PREFIX.ffmpeg9}/lib/ffmpeg9/lib${lib:tl}.${SHLIB_EXT}
 .endfor
 
 # \todo Recheck if this is needed.
@@ -262,7 +261,7 @@
 # 2024-01: 1.36 works with mariadb 11.2 on GNU/Linux
 # 2026-07: 1.38 works with mariadb 10.6 on NetBSD
 DEPENDS+=			p5-DBD-MariaDB-[0-9]*:../../databases/p5-DBD-MariaDB
-MYSQL_VERSIONS_ACCEPTED=	mariadb118 mariadb114 mariadb1011 mariadb106 mariadb105
+MYSQL_VERSIONS_ACCEPTED=	mariadb118
 CMAKE_CONFIGURE_ARGS+=		-DZM_DB_TYPE=MariaDB
 .endif
 .if !empty(PKG_OPTIONS:Mmysql)
@@ -417,7 +416,7 @@
 .include "../../archivers/bzip2/buildlink3.mk"
 # \todo Consider graphics/libv4l.
 .include "../../devel/glib2/buildlink3.mk"
-.include "../../devel/pcre/buildlink3.mk"
+.include "../../devel/pcre2/buildlink3.mk"
 .include "../../devel/zlib/buildlink3.mk"
 .include "../../net/libvncserver/buildlink3.mk"
 .include "../../multimedia/mp4v2/buildlink3.mk"
